File: java/src/test/java/org/my_projects/skippy/kangaroo/KangarooTest.java
package org.my_projects.skippy.kangaroo;

import org.junit.jupiter.api.Test;
import org.my_projects.skippy.grid.Grid;
import org.my_projects.skippy.grid.GridItem;
import org.my_projects.skippy.grid.Point;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class KangarooTest {

    @Test
    void chortle_ShouldPrintToStdout() {

        // GIVEN
        var testOutStream = new ByteArrayOutputStream();
        var systemOutStream = System.out;

        // WHEN
        System.setOut(new PrintStream(testOutStream));
        new Kangaroo().chortle("Hi there!");
        System.setOut(systemOutStream);

        // THEN
        assertTrue(testOutStream.toString().matches("Hi there!\\s*"));
    }

    @Test
    void findHome_ShouldEventuallyFindHome() throws Exception {

        // GIVEN
        var grid = new Grid(5);
        var kangaroo = new Kangaroo();
        var home = new GridItem() {};
        grid.placeItem(kangaroo, new Point(0, 0));
        grid.placeItem(home, new Point(4, 4));

        // WHEN
        kangaroo.findHome(grid, home);

        // THEN
        assertEquals(grid.getItemLocation(home), grid.getItemLocation(kangaroo));
    }
}

File: ruby/test/kangaroo_test.rb
# frozen_string_literal: true

require './test/test_helper'
require 'skippy/kangaroo'

describe Kangaroo do
  it 'should print to stdout when chortling' do
    out, = capture_io do
      Kangaroo.new.chortle('Hi there!')
    end
    _(out).must_match(/Hi there!\s*/)
  end

  it 'should eventually find home' do
    # GIVEN
    grid = Grid.new(5)
    kangaroo = Kangaroo.new
    home = GridItem.new
    grid.place_item(kangaroo, Point.new(0, 0))
    grid.place_item(home, Point.new(4, 4))

    # WHEN
    kangaroo.find_home(grid, home)

    # THEN
    _(grid.get_item_location(kangaroo)).must_equal(grid.get_item_location(home))
  end
end
